tarofreighter/chain_porter.go

```diff
@@ -501,6 +520,38 @@ func (p *ChainPorter) transferReceiverProof(pkg *sendPackage) error {
 	return nil
 }
 
+// calcDeliveryDelay calculates a time delay with a duration based on the
+// previous parcel delivery attempt timestamps.
+func calcDeliveryDelay(timestamps []time.Time) time.Duration {
+	// If there are no previous parcel delivery attempts, then we'll
+	// return early.
+	if len(timestamps) == 0 {
+		return time.Duration(0)
+	}
+
+	// Otherwise we'll select the latest timestamp and compute a relative
+	// delay.
+
+	// Get the latest timestamp without assuming order.
+	latestTimestamp := timestamps[0]
+	for _, timestamp := range timestamps {
+		if timestamp.After(latestTimestamp) {
+			latestTimestamp = timestamp
+		}
+	}
+
+	// Compute the delay duration relative to the latest attempt timestamp
+	// and current time.
+	baseDelay := time.Duration(time.Minute * 3)
```

Comment:
This would probably make a good package level constant.

Reply:
Moved into config now.